### PR TITLE
add prefer const rule

### PR DESCRIPTION
**Pull request by DarylThayil**
Add prefer const rule and fix for vars never reassigned

### CODE REVIEW

File: lib/msal-core/src/UserAgentApplication.ts [hunk 1]
```diff
@@ -1474,7 +1474,7 @@ export class UserAgentApplication {
   /* tslint:disable:no-string-literal */
   private saveAccessToken(response: AuthResponse, authority: string, parameters: any, clientInfo: string, idTokenObj: IdToken): AuthResponse {
       let scope: string;
-      let accessTokenResponse = { ...response };
+      const accessTokenResponse = { ...response };
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: On line 1510, we assign new values to accessTokenResponse. We can't make this a const.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 2]
```diff
@@ -1337,19 +1337,19 @@ export class UserAgentApplication {
...
+              const idTokenObj = new IdToken(accessTokenCacheItem.value.idToken);
               if (!account) {
                   account = this.getAccount();
                   if (!account) {
                       throw AuthError.createUnexpectedError("Account should not be null here.");
                   }
               }
               const aState = this.getAccountState(serverAuthenticationRequest.state);
-              let response : AuthResponse = {
+              const response : AuthResponse = {
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: We can't make this const, in setResponseIdToken we modify the idToken field.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 1 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: Const only restricts the memory reference of the top level item so in this case an object is created with a memory reference

When we assign values to the references inside that object we are not changing the memory reference of the accessTokenResponse 
```js
const accessTokenResponse = {};
accessTokenResponse = {} // not allowed
acessTokenResponse.foo = 'bar' //allowed
```

File: lib/msal-core/src/UserAgentApplication.ts (hunk 2 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: see previous response

File: lib/msal-core/src/UserAgentApplication.ts [hunk 3]
```diff
@@ -587,7 +587,7 @@ export class UserAgentApplication {
 
             const responseType = this.getTokenType(account, request.scopes, true);
 
-            let serverAuthenticationRequest = new ServerRequestParameters(
+            const serverAuthenticationRequest = new ServerRequestParameters(
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: if we make this const, does changing fields in the object throw an error?

Comment (pkanher617) [reply lib/msal-core/src/UserAgentApplication.ts]: I believe we change fields in some of the functions.

File: lib/msal-core/src/ServerRequestParameters.ts [hunk 4]
```diff
@@ -311,7 +311,7 @@ export class ServerRequestParameters {
    * @param request
    */
     private sanitizeEQParams(request: AuthenticationParameters) : StringDict {
-        let eQParams : StringDict = request.extraQueryParameters;
+        const eQParams : StringDict = request.extraQueryParameters;
```
Comment (pkanher617) [inline lib/msal-core/src/ServerRequestParameters.ts]: We modify eqParams down below, we can't make this const.

**Pull request merged**